mteb/models/openai_models.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import logging
+from functools import partial
+from typing import Any
+
+import numpy as np
+import torch
+
+from mteb.import_utils import requires_package
+from mteb.model_meta import ModelMeta
+from mteb.models.utils import corpus_to_texts
+
+logger = logging.getLogger(__name__)
+
+
+class OpenAIWrapper:
+    def __init__(self, model_name: str, embed_dim: int | None = None) -> None:
+        requires_package(self, "openai", "Openai text embedding")
+        from openai import OpenAI
+
+        self._client = OpenAI()
+        self._model_name = model_name
+        self._embed_dim = embed_dim
+
+    def encode(self, sentences: list[str], **kwargs: Any) -> torch.Tensor | np.ndarray:
+        requires_package(self, "openai", "Openai text embedding")
+        from openai import NotGiven
+
+        if self._model_name == "text-embedding-ada-002" and self._embed_dim is not None:
+            logger.warning(
+                "Reducing embedding size available only for text-embedding-3-* models"
+            )
+
+        return self._to_numpy(
+            self._client.embeddings.create(
+                input=sentences,
+                model=self._model_name,
+                encoding_format="float",
+                dimensions=self._embed_dim or NotGiven(),
+            )
+        )
+
+    def encode_queries(
+        self, queries: list[str], **kwargs: Any
+    ) -> torch.Tensor | np.ndarray:
+        return self.encode(["query: " + q for q in queries], **kwargs)
+
+    def encode_corpus(
+        self, corpus: list[dict[str, str]] | dict[str, list[str]], **kwargs: Any
+    ) -> torch.Tensor | np.ndarray:
+        sentences = corpus_to_texts(corpus)
+        return self.encode(sentences, **kwargs)
+
+    def _to_numpy(self, embedding_response) -> np.ndarray:
+        return np.array([e.embedding for e in embedding_response.data])
+
+
+text_embedding_3_small = ModelMeta(
+    name="text-embedding-3-small",
+    revision="1",
+    release_date="2024-01-25",
+    languages=["eng-Latn"],
```

Comment:
```suggestion
    languages=None, # supported languages not specified
```
assuming it is not specified